lib/child_process.js

```diff
@@ -60,22 +60,29 @@ function stdioStringToArray(option) {
 }
 
 exports.fork = function fork(modulePath /* , args, options */) {
-
   // Get options and args arguments.
   var execArgv;
-  var options = {};
-  var args = [];
-  var pos = 1;
-  if (pos < arguments.length && Array.isArray(arguments[pos])) {
-    args = arguments[pos++];
-  }
 
-  if (pos < arguments.length && arguments[pos] != null) {
-    if (typeof arguments[pos] !== 'object') {
-      throw new ERR_INVALID_ARG_VALUE(`arguments[${pos}]`, arguments[pos]);
-    }
+  var args = arguments[1];
+  var options = arguments[2];
```

Comment:
Accessing arguments without checking the length can lead to OOB accesses. That is not that bad anymore but that requires quite some work for the compiler and it slows things down. Therefore this way of doing it is not really good.

Reply:
would this be better - 
```js
var pos = 1;

  var args = pos < arguments.length ? arguments[pos++] : [];
  var options = pos < arguments.length ? arguments[pos++] : {};
```

Reply:
Likely. Just try to make sure each code path is not hit more than absolutely necessary.